grpcclient/middleware.go

```diff
@@ -0,0 +1,123 @@
+package grpcclient
+
```

Comment:
I think the metadata constants and some of the functions are not naturaly part of the client module

I have a middleware package in my branch of go-rkvstcommon for things like this

https://github.com/rkvst/go-rkvstcommon/tree/dev/robin/8337-common-metadata/middleware

I think grpcclient is fine in general for things that are definitely only the concern of client users and implementors